### PR TITLE
Make GitHub Actions build on Linux (to detect and prevent future build regressions)

### PR DESCRIPTION
**Pull request by hartwork**
This sits on top of pull requests #67, #69, #70, #71 to have a chance for a green CI. The idea is to first get all those merged, and then I would rebase this onto latest `master` to cut it down to the essence which is adding CI.

### CODE REVIEW

**Comment by hartwork** [comment]
Rebased onto latest `master` now. PS: This very Ci can be seen in action at https://github.com/hartwork/ttyplot/actions .

**Comment by hartwork** [comment]
@tenox7 rebased, only sits on top of #67 now.

**Comment by hartwork** [comment]
@tenox7 do you have a minute for review?

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@ DESTDIR   ?=
 PREFIX    ?= /usr/local
 MANPREFIX ?= $(PREFIX)/man
 CFLAGS += -Wall -Wextra
-LDLIBS += -lcurses -ltinfo
+LDLIBS += `pkg-config --libs ncurses 2>/dev/null || echo '-lcurses -ltinfo'`
```
Comment (tenox7) [inline Makefile:5]: this will not work on a system that doesn't have pkg-config

Comment (hartwork) [reply Makefile:5]: @tenox7 how will it not? That's what the `||` is for.

PS: Having a build require pkg-config would not be something crazy to ask from users also. It's about as standard of a build tool as it gets.

**Comment by hartwork** [comment]
Rebased off of #67 and onto latest `master` now.

**Comment by hartwork** [comment]
@tenox7 we're in three weeks here now without any real progress. If this pull request is not welcome, now would be a good time to say, and we'll just close it and I'll move on. How would you like to proceed?

**Comment by tenox7** [comment]
PRs are most welcome. I'm mostly busy with other projects and I didn't want to risk breaking things. Never used github actions before so not sure what to expect from this. I'll merge it.

**Pull request merged**